firefox_ui_tests/functional/locationbar/test_go_button.py

```diff
@@ -0,0 +1,36 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+from marionette_driver import By
+
+
+class TestGoButton(FirefoxTestCase):
+    """ This replaces
+    This replaces http://hg.mozilla.org/qa/mozmill-tests/file/default/firefox/tests/functional/testAwesomeBar/testGoButton.js
+    Check a go button displays after text has been added to the awesome bar, takes user to correct results.
+    """
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+    def test_gobutton(self):
+        # no input, go button should not display
+        go_button = self.browser.navbar.go_button
+        self.assertFalse(go_button.is_displayed())
```

Comment:
The `is_displayed()` method is not working reliable and we got told to not use it. So here it would be better to use `button.get_attribute('foobar')` whereby foobar is the name of the js property which returns the visibility state of the button.